feat: added boost_fee param

marcellorigotti opened the pull request:
# Pull Request

Closes: PRO-1146
## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have updated documentation where appropriate.

## Summary

Added boost_fee param to the broker/lp CLI/API when opening a deposit channel 
-it's an optional parameter that defaults to 0 if not provided
-Added boost_fee also to the relevant extrinsics (request_liquidity_deposit_address and request_swap_deposit_address)
Currently the extrinsics use this new parameter only in the events they emit (SwapDepositAddressReady, LiquidityDepositAddressReady) -> This is a breaking change for the CFE.

msgmaxim commented on api/bin/chainflip-broker-api/src/main.rs:
```diff
@@ -54,6 +54,7 @@ pub trait Rpc {
 		destination_asset: RpcAsset,
 		destination_address: String,
 		broker_commission_bps: BasisPoints,
+		boost_fee: Option<BasisPoints>,
```
Do we have tests to check that we don't break compatibility between releases @dandanlen? I know some bounder tests use SDK, but I'm not sure if they, for example, use `channel_metadata`, so it would be good to have more explicit/exhaustive tests (ideally using hardcoded commands with curl or something like that).

kylezs replied:
Yeah I was wondering in that case do we want to break compatibility or create a new call, and deprecate the old one

msgmaxim replied:
I was thinking simply putting boost_fee as the last parameter would preserve comatibility (you can use `null` to indicate that `channel_metadata` should be parsed into `None`).

marcellorigotti commented:
Addressed the comments, just one question about the documentation, I created a ticket to not forget to update it (https://linear.app/chainflip/issue/PRO-1168/documentation-include-boost-fee-param) should I go on and do it now, or do we want to wait until we release it before updating the docs?
Is there some common practice here?

kylezs commented on state-chain/pallets/cf-swapping/src/lib.rs:
```diff
@@ -477,6 +478,7 @@ pub mod pallet {
 			destination_asset: Asset,
 			destination_address: EncodedAddress,
 			broker_commission_bps: BasisPoints,
+			boost_fee: BasisPoints,
 			channel_metadata: Option<CcmChannelMetadata>,
```
we should have the order be consistent with the broker

kylezs reviewed (commented):
We won't merge this until release/1.2 is branched

martin-chainflip commented:
@acdibble FYI, there's an event change in this PR.
